agent/src/main/java/com/microsoft/applicationinsights/agent/internal/coresync/impl/ImplementationsCoordinator.java

```diff
@@ -49,17 +49,16 @@
     private volatile long maxSqlMaxQueryThresholdInMS = 10000L;
     private volatile long redisThresholdInNS = 10000L * 1000000;
 
-    private static RuntimeExceptionDecider runtimeExceptionDecider;
+    private final RuntimeExceptionDecider runtimeExceptionDecider = new RuntimeExceptionDecider();
 
     private static ConcurrentHashMap<String, RegistrationData> notificationHandlersData = new ConcurrentHashMap<String, RegistrationData>();
```

Comment:
It appears this can be a `final` instance variable also.

Reply:
`public void setExceptionData (DataOfConfigurationForException exceptionData)` this method is called to set the exceptiondata on `runtimeExceptionDecider` from `AgentImplementation.java` hence this cannot be instance variable. We need to keep that as class variable.

Reply:
I was referring to `notificationHandlersData`. 

Since this is already a static singleton, the result is the same if it's declared as static or not.

Reply:
@littleaj Apologies for the confusion. This one is not being used at all in the new code path. This was being used by `public String register(ClassLoader classLoader, AgentNotificationsHandler handler)` which is now deprecated. Moot point to change this at the moment.

Reply:
ah, yes. that is true